scripts/test_examples.py

```diff
@@ -273,86 +275,86 @@ def __init__(self, args):
         self.build_flags = ['setup.py', 'build', '--translate']
         self.init_submodule()
 
-    def __del__(self):
+    def __del__(self) -> None:
         self.deinit_submodule()
 
-    def gen_cc_db(self):
+    def gen_cc_db(self) -> None:
         pass
 
-    def transpile(self):
+    def transpile(self) -> None:
         pass
 
-    def build(self):
+    def build(self) -> None:
         with pb.local.cwd(self.repo_dir):
             invoke(python, *self.build_flags)
 
-    def test(self):
+    def test(self) -> None:
         with pb.local.cwd(self.repo_dir):
             invoke(python, ['setup.py', 'test'])
 
 class TinyCC(Test):
-    def __init__(self, args):
+    def __init__(self, args: argparse.Namespace) -> None:
         self.args = args
         self.project_name = 'tinycc'
         self.transpiler_args = []
         self.ib_cmd = ['make', '-j{}'.format(NUM_JOBS)]
         self.example_dir = build_path(
             c.EXAMPLES_DIR, self.project_name, is_dir=True)
         self.repo_dir = build_path(self.example_dir, 'repo', is_dir=True)
-        self.build_flags = []
+        self.build_flags: List[str] = []
         self.init_submodule()
         self.rust_src = os.path.join(self.repo_dir, 'rust')
 
-    def __del__(self):
+    def __del__(self) -> None:
         self.deinit_submodule()
 
-    def autotools(self):
+    def autotools(self, configure_args: List[str] = []) -> None:
```

Comment:
Default argument can't be mutable, like `[]`.

Reply:
Used immutably, this is fine.